### PR TITLE
feat: Add spinner for ticket quantity

### LINKED ISSUE #194: Ticket item should have a spinner in qty range
Currently  the range is shown in textview it should be shown in a spinner.
The ticket whose spinner gets selected should be passed on to the next fragment when the user clicks on the register button

### PR DESCRIPTION
**Pull request by simarsingh24**
fixes #194

### CODE REVIEW

**Comment by simarsingh24** [comment]
SS after change ->
image
image

File: app/src/main/java/org/fossasia/openevent/general/ticket/TicketViewHolder.kt [hunk 1]
```diff
@@ -1,17 +1,32 @@
...
 class TicketViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
 
-    fun bind(ticket: Ticket) {
+    fun bind(ticket: Ticket, selectedListener: TicketSelectedListener?) {
         itemView.ticketName.text = ticket.name
 
         if (!ticket.minOrder.isNullOrEmpty() && !ticket.maxOrder.isNullOrEmpty()) {
-            itemView.orderRange.text ="${ticket.minOrder}-${ticket.maxOrder}"
+            val spinnerList = ArrayList<String>()
+            for (i in ticket.minOrder!!.toInt()..ticket.maxOrder!!.toInt()) {
```
[the remark this answers is not in the document]
Comment (nikit19) [reply app/src/main/java/org/fossasia/openevent/general/ticket/TicketViewHolder.kt]: @simarsingh24 avoid using `!!` they throw null pointer exceptions at run time if null value is encountered

**Comment by simarsingh24** [comment]
@iamareebjamal please review

File: app/src/main/java/org/fossasia/openevent/general/ticket/TicketViewHolder.kt [hunk 2]
```diff
@@ -1,17 +1,32 @@
...
 class TicketViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
 
-    fun bind(ticket: Ticket) {
+    fun bind(ticket: Ticket, selectedListener: TicketSelectedListener?) {
         itemView.ticketName.text = ticket.name
 
         if (!ticket.minOrder.isNullOrEmpty() && !ticket.maxOrder.isNullOrEmpty()) {
-            itemView.orderRange.text ="${ticket.minOrder}-${ticket.maxOrder}"
+            val spinnerList = ArrayList<String>()
+            for (i in (ticket.minOrder?.toInt() ?: 0)..(ticket.maxOrder?.toInt() ?: 0)) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/ticket/TicketViewHolder.kt]: Make min and max order int

**Comment by simarsingh24** [comment]
PR updated

File: app/src/main/java/org/fossasia/openevent/general/ticket/Ticket.kt [hunk 3]
```diff
@@ -24,7 +24,7 @@ data class Ticket(
         val description: String?,
         val type: String?,
         val name: String,
-        val maxOrder: String?,
+        val maxOrder: Int? = null,
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/ticket/Ticket.kt]: Make non-nullable and initialize to 0

Comment (simarsingh24) [reply app/src/main/java/org/fossasia/openevent/general/ticket/Ticket.kt]: done

**Pull request merged**